### PR TITLE
DOC: mention PyArray_GetField steals a reference

### PR DESCRIPTION
**Pull request by SahanaJavgal**
Updates Document mentions that PyArray_GetField steals a reference to its descr argument
along with the meaning of steals to reference.

### CODE REVIEW

File: doc/source/reference/c-api.array.rst [hunk 1]
```diff
@@ -288,13 +288,6 @@ From scratch
     are passed in they must be consistent with the dimensions, the
     itemsize, and the data of the array.
 
-.. c:function:: PyObject* PyArray_SimpleNew(int nd, npy_intp* dims, int typenum)
-
-    Create a new uninitialized array of type, *typenum*, whose size in
-    each of *nd* dimensions is given by the integer array, *dims*.
-    This function cannot be used to create a flexible-type array (no
-    itemsize given).
-
```
Comment (mattip) [inline doc/source/reference/c-api.array.rst]: Did you mean to delete this?

File: doc/source/reference/c-api.array.rst [hunk 2]
```diff
@@ -1660,10 +1653,11 @@ Conversion
...
 
-    Equivalent to :meth:`ndarray.getfield<numpy.ndarray.getfield>` (*self*, *dtype*, *offset*). Return
-    a new array of the given *dtype* using the data in the current
-    array at a specified *offset* in bytes. The *offset* plus the
-    itemsize of the new array type must be less than *self*
+    Equivalent to :meth:`ndarray.getfield<numpy.ndarray.getfield>` (*self*, *dtype*, *offset*). This function 
+    :meth:`steals a reference<https://stackoverflow.com/questions/23644926/what-is-reference-stealing-and-borrowing>`
+    to ``PyArray_Descr`` and return a new array of the given *dtype* 
+    using the data in the current array at a specified *offset* in bytes. 
+    The *offset* plus the itemsize of the new array type must be less than *self*
```
Comment (mattip) [inline doc/source/reference/c-api.array.rst]: Needs to be reformatted to 80 columns. You can break on any blank space. 

Missing a space between reference and `<`. The `:meth:` is not needed. Perhaps the pytohn documentation is better than a stack overflow question. `https://docs.python.org/3/c-api/intro.html?reference-count-details`

`and return a new array` -> `and returns a new array`

File: doc/source/reference/c-api.array.rst (hunk 1 above)
Comment (mattip) [reply doc/source/reference/c-api.array.rst]: Ahh, this is #12261. This should remain, the other document should reference this one.

**Comment by mattip** [comment]
Fixes #10627 and  makes progress with #12261

**Comment by mattip** [comment]
It is not part of the issue, but as long as we are touching this, all the text enclosed in asterisks like`*offset*` should be reformatted. If they are function parameters, they need a single back-tick, otherwise they should have double-backticks.

File: doc/source/reference/c-api.array.rst (hunk 1 above)
Comment (SahanaJavgal) [reply doc/source/reference/c-api.array.rst]: Noted, Thanks

File: doc/source/reference/c-api.array.rst (hunk 2 above)
Comment (mattip) [reply doc/source/reference/c-api.array.rst]: Maybe I was unclear. I was suggesting
```
`steals a reference <https://docs.python.org/3/c-api/intro.html?reference-count-details>`
```
not `**steals a reference**` which will simply be rendered in bold typeface.

**Comment by kritisingh1** [comment]
xref #13649 we can close this @mattip

**Pull request closed without merging**